pyro/infer/mcmc/mcmc.py

```diff
@@ -162,10 +167,23 @@ def _traces(self, *args, **kwargs):
                 if isinstance(val, Exception):
                     # Exception trace is already logged by worker.
                     raise val
-                elif val is not None:
-                    yield val
-                else:
+                results_buffer[chain_id - 1].append(val)
+                if val is None:
                     active_workers -= 1
+                while results_buffer[cur_idx]:
+                    val = results_buffer[cur_idx].popleft()
+                    if val:
```

Comment:
what does `if val` mean here? I.e. what types and values can `val` take on? Can you more precisely test `if val is not None`?

Reply:
This isn't needed - I will remove it.